pinata/pinata.go

```diff
@@ -0,0 +1,225 @@
+		}
+	}
+
+	url := "https://api.pinata.cloud/pinning/pinFileToIPFS"
+	method := "POST"
+
+	payload := &bytes.Buffer{} // buffer writer
+	writer := multipart.NewWriter(payload)
+	file, fileOpenError := os.Open(path)
+
+	if fileOpenError != nil {
+		return nil, &errors.BifrostError{
+			Err:       fileOpenError,
+			ErrorCode: errors.ErrFileOperationFailed,
+		}
+	}
+
+	part1, err := writer.CreateFormFile("file", filepath.Base(path))
+	if err != nil {
+		return nil, &errors.BifrostError{
+			Err:       fmt.Errorf("file does not exist: %s", path),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+
+	_, errFile1 := io.Copy(part1, file)
+	if errFile1 != nil {
+		return nil, &errors.BifrostError{
+			Err:       fmt.Errorf("failed to copy multipart data"),
+			ErrorCode: errors.ErrBadRequest,
+		}
+	}
+
+	defer file.Close()
+
+	 metaData := map[string]string{"name": filename}
+	 pinataMetadata,err := json.Marshal(metaData)
+	 if err!= nil {
+		 return nil, &errors.BifrostError{
+        Err:       fmt.Errorf("failed to marshal metadata"),
+        ErrorCode: errors.ErrBadRequest,
+      }
+	 }
+
+	 pinataOptions,err := json.Marshal(options)
+
+
+	 if err!= nil {
+		 return nil, &errors.BifrostError{
+        Err:       fmt.Errorf("failed to convert metadata to JSON"),
+        ErrorCode: errors.ErrBadRequest,
+      }
+	 }
+
+  _ = writer.WriteField("pinataOptions", string(pinataOptions))
+  _ = writer.WriteField("pinataMetadata", string(pinataMetadata))
+
+	err = writer.Close()
+  if err != nil {
+    return nil, &errors.BifrostError{
+      Err:       fmt.Errorf("failed to close multipart writer"),
+      ErrorCode: errors.ErrBadRequest,
+    }
+  }
+
+	client := &http.Client {
+  }
+  req, err := http.NewRequest(method, url, payload)
+
+  if err != nil {
+    return nil, &errors.BifrostError{
+			Err:       err,
+      ErrorCode: errors.ErrBadRequest,
+		}
+  }
+	 req.Header.Add("Authorization", "Bearer " + p.Config().PinataJWT)
+	 
+
+  req.Header.Set("Content-Type", writer.FormDataContentType())
+  res, err := client.Do(req)
+	fmt.Println(req.Body)
+
+	if err != nil {
+    return nil, &errors.BifrostError{
+      Err:       err,
+      ErrorCode: errors.ErrBadRequest,
+    }
+  }
+  defer res.Body.Close()
+  body, err := io.ReadAll(res.Body)
+
+	 if err != nil {
+    return nil, &errors.BifrostError{
+			Err:       err,
+      ErrorCode: errors.ErrBadRequest,
+		}
+  }
+	var responseCode int = res.StatusCode
+	fmt.Println(responseCode)
+
+	if responseCode != 200 {
+		return nil, &errors.BifrostError{
+      Err:       fmt.Errorf("upload request failed with status %d", responseCode),
+      ErrorCode: errors.ErrBadRequest,
+    }
+	}
+
+		res_ := SuccessResponse{}
+    json.Unmarshal([]byte(body), &res_)
+
+		return &types.UploadedFile{
+			Size: res_.PinSize,
+			Name: res_.IpfsHash,
+			Preview: "https://gateway.pinata.cloud/ipfs/"+res_.IpfsHash ,
+	}, nil
+
+}
+
+/*
+
+ */
+func (g *PinataIPFSStorage) Disconnect() error {
```

Comment:
Actually....this disconnect can do something tangible but that would mean that our config needs to have an actual `Client` attached to it. I'll explain...

You see all the gymnastics 🙃 above needed for creating an HTTP client to make a request....we can bundle that as the client so we don't have to recreate it all the time. 
That means you need to move it into the `newPinataCloud` function when creating the rainbow bridge,

Reply:
Maybe you should take care of this setup, i have it rough in my head and might implement it different from what's in your head.. we can make this a new PR for a new issue.

Reply:
Sure thing. 
Let's do this way....once you're done fixing other comments that don't relate to this one... please let me know then I'll pull your branch and update it with this changes.

Reply:
I already created a new ``request.go`` file, can you make those changes in there? in a new PR

Reply:
Great. Thank you. I'll review the new changes when I get home. Currently replying via mobile 🙏